assets/js/search.js

```diff
@@ -30,17 +29,28 @@ setDateToday();
     )
 })(jQuery);
 
-$(document).ready(function CheckPreviousLocation() {
-  const elem = document.getElementById("location");
-  if (sessionStorage.getItem("locationPerm")) {
-    console.log(
-      `Set the Location based on PageLoad...` +
-        sessionStorage.getItem("locationPerm")
-    );
-    elem.value = sessionStorage.getItem("locationPerm");
-  }
+document.getElementById("logout-confirm-btn").addEventListener(
+    "click", logOut
+)
+
+const FORM = document.getElementById("main-form");
+const STORAGE_ITEM = "location";
+const LOCATION_INPUT = document.getElementById("location");
+$(document).ready(() => {
+    const val = sessionStorage.getItem(STORAGE_ITEM);
+    if (val) {
+        console.log(`Set the Location based on PageLoad...` + val);
+        LOCATION_INPUT.value = val;
+    }
 });
 
```

Comment:
@weihesdlegend so the function `fetchLocation` is being moved here? looks good, just need to make sure if that works

Reply:
Previously you used `onChange` to track modifications to the location input element. However it misses auto complete as well as find my location. So the change in this PR is a completely new approach.

Reply:
by `find my` location, do you mean the arrow sign functionality that is on the right top of the price bar ? 
- I checked that is working on production on the web ?